integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiFmwAuxiliaryImageSample.java

```diff
@@ -0,0 +1,385 @@
+   * specialized for JRF, but is otherwise similar to
+   * the WLS initial use case.
+   * @see ItMiiSample#testWlsInitialUseCase for more...
+   */
+  @Test
+  @Order(1)
+  @DisabledIfEnvironmentVariable(named = "SKIP_JRF_SAMPLES", matches = "true")
+  @DisplayName("Test to verify MII sample JRF initial use case using auxiliary image")
+  public void testAIFmwInitialUseCase() {
+    callFmwInitialUseCase();
+  }
+
+  /**
+   * Test to verify JRF update1 use case using auxiliary image.
+   * @see ItMiiSample#testWlsInitialUseCase for more...
+   */
+  @Test
+  @Order(2)
+  @DisabledIfEnvironmentVariable(named = "SKIP_JRF_SAMPLES", matches = "true")
+  @DisplayName("Test to verify MII sample JRF update1 use case using auxiliary image")
+  public void testAIFmwUpdate1UseCase() {
+    callFmwUpdate1UseCase();
+  }
+
+  /**
+   * Test to verify JRF update2 use case using auxiliary image.
+   * @see ItMiiSample#testWlsInitialUseCase for more...
+   */
+  @Test
+  @Order(3)
+  @DisabledIfEnvironmentVariable(named = "SKIP_JRF_SAMPLES", matches = "true")
+  @DisplayName("Test to verify MII sample JRF update2 use case using auxiliary image")
+  public void testAIFmwUpdate2UseCase() {
+    callFmwUpdate2UseCase();
+  }
+
+  /**
+   * Test to verify JRF update3 use case using auxiliary image.
+   * @see ItMiiSample#testWlsInitialUseCase for more...
+   */
+  @Test
+  @Order(4)
+  @DisabledIfEnvironmentVariable(named = "SKIP_JRF_SAMPLES", matches = "true")
+  @DisplayName("Test to verify MII sample JRF update3 use case using auxiliary image")
+  public void testAIFmwUpdate3UseCase() {
+    callFmwUpdate3UseCase();
+  }
+
+  /**
+   * Test to verify WLS update4 use case using auxiliary image.
+   * Update Work Manager Min and Max Threads Constraints via a configmap and updates the
+   * domain resource introspectVersion.
+   * Verifies the sample application is running
+   * and detects the updated configured count for the Min and Max Threads Constraints.
+   */
+  @Test
+  @Order(5)
+  @DisabledIfEnvironmentVariable(named = "SKIP_JRF_SAMPLES", matches = "true")
+  @DisplayName("Test to verify MII sample JRF update4 use case using auxiliary image")
+  public void testAIFmwUpdate4UseCase() {
+    callFmwUpdate4UseCase();
+  }
+
+  /**
+   * Delete DB deployment and Uninstall traefik.
+   */
+  @AfterAll
+  public void tearDownAll() {
+    // db cleanup or deletion
+    if (envMap != null) {
+      logger.info("Running samples DB cleanup");
+      Command.withParams(new CommandParams()
+          .command(MII_SAMPLES_SCRIPT + " -precleandb")
+          .env(envMap)
+          .redirect(true)).execute();
+    }
+
+    // uninstall traefik
+    if (traefikNamespace != null) {
+      logger.info("Uninstall traefik");
+      Command.withParams(new CommandParams()
+          .command("helm uninstall traefik-operator -n " + traefikNamespace)
+          .redirect(true)).execute();
+    }
+  }
+
+  private static void assertImageExistsAndPushIfNeeded() {
```

Comment:
there is duplicated code in all three test classes, can you move them out of the It class and make calls?

Reply:
+1

Reply:
This method is not the only duplicate code. It looks like constants, DomainType, initAll, exectTest*, callFmw*, have all bean duplicated. 

Suggestion: create one class that contains the complete implementation and call it ITMiiSampleHelper or somesuch, then create four very similar small test suite classes 'ItMIISample[WLS|JRF][Main|Aux]Image' that (a) set a modal value on the helper class to change its behavior between WLS|JRF and Main|Aux, and (b) call the helper tests.  The only difference between the four need only be description text, class name, and the setting of the modal...

Reply:
changed

Reply:
There's remaining duplicate code between the four tests in 'init' and 'tearDown'.  These can invoke a 'universal' init and tearDown in the new helper class. Once this duplicate code is factored out, then most of the constants and imports in the four tests no longer need to be defined.

Also, you can add a `auxiliaryImageEnabled()` method that returns `true` or `false` - and use it to replace repeat usage of `String useAuxiliaryImage = (getImageType().equalsIgnoreCase(ImageType.AUX.toString()))  ? "true" : "false";`

Reply:
changed